core/domain/user_services.py

```diff
@@ -896,6 +898,71 @@ def _update_user_contribution_rights(user_contribution_rights):
         remove_contribution_reviewer(user_contribution_rights.id)
 
 
+def _update_reviewer_counts_in_community_contribution_stats_transactional(
+        future_user_contribution_rights):
+    """Updates the reviewer counts in the community contribution stats based
+    on the updates to the given user contribution rights.
+
+    Args:
+        future_user_contribution_rights: UserContributionRights. The most up to
+            date user contribution rights.
+    """
+    past_user_contribution_rights = get_user_contribution_rights(
+        future_user_contribution_rights.id)
+    stats_model = suggestion_models.CommunityContributionStatsModel.get()
+
+    future_languages_that_reviewer_can_review = set(
+        future_user_contribution_rights
+        .can_review_translation_for_language_codes)
+    past_languages_that_reviewer_can_review = set(
+        past_user_contribution_rights.can_review_translation_for_language_codes)
+
+    languages_that_reviewer_can_no_longer_review = (
+        past_languages_that_reviewer_can_review.difference(
+            future_languages_that_reviewer_can_review))
+    new_languages_that_reviewer_can_review = (
+        future_languages_that_reviewer_can_review.difference(
+            past_languages_that_reviewer_can_review))
+
+    # Update question reviewer counts.
+    if past_user_contribution_rights.can_review_questions and not (
+            future_user_contribution_rights.can_review_questions):
+        stats_model.question_reviewer_count -= 1
+    if not past_user_contribution_rights.can_review_questions and (
+            future_user_contribution_rights.can_review_questions):
+        stats_model.question_reviewer_count += 1
+    # Update translation reviewer counts.
+    for language_code in languages_that_reviewer_can_no_longer_review:
+        stats_model.translation_reviewer_counts_by_lang_code[
+            language_code] -= 1
```

Comment:
Is it worth removing from the array if it reaches 0 (for symmetry with the addition case below)?

Reply:
Done!